e2e/_suites/fleet/fleet.go

```diff
@@ -89,7 +89,8 @@ func (fts *FleetTestSuite) afterScenario() {
 
 	serviceName := common.ElasticAgentServiceName
 
-	if fts.InstallerType != "" {
+	// if DEVELOPER_MODE=true let's not uninstall/unenroll the agent
```

Comment:
Dummy question, is `fts.InstallerType != ""` expected to be empty? just in case it should be added to the condition like the below pseudocode 


```
	if !common.DeveloperMode || fts.InstallerType != "" {
```

Reply:
I had the same question that you and I took a look at the code: we are always setting that value, so it's never empty.

Reply:
@v1v you were totally right. There is only one scenario that is not using the installerType, therefore the protection code should be there as you proposed. Fixing it now.

Reply:
BTW, I considered using this block instead:

```golang
if !common.DeveloperMode && fts.InstallerType != "" {
```

Because we want to satisfy both conditions at the same time:

| A = `dev_mode` | B=`InstallerType != ""` | f(x) |
|-----------|------------------|-----|
| 0 | 0 | 0 |
| 0 | 1 | 1 |
| 1 | 0 | 0 |
| 1 | 1 | 0 |

f(x) = !A*B